src/Simulation/Native/src/simulator/wavefunction.hpp

```diff
@@ -549,6 +550,40 @@ class Wavefunction
         return kernels::jointprobability(wfn_, bs, get_qubit_positions(qs));
     }
 
+    /// \pre: Qubits, listed in `q`, must be unentangled and in state |0>.
+    /// place these `n` qubits into superposition of 2^n basis vectors with (re, im) amplitudes, where the order of
+    /// qubits in array `q` defines (big or little endian?) order of the basis vectors.
+    void inject_state(const std::vector<logical_qubit_id>& qubits, const std::vector<ComplexType>& amplitudes)
+    {
+        assert((static_cast<size_t>(1) << qubits.size()) == amplitudes.size());
+
+        if (num_qubits_ != qubits.size()) throw std::runtime_error("Subsystem state injection not supported (yet)");
+
+        flush();
+
+        // Check prerequisites.
+        for (logical_qubit_id q : qubits)
+        {
+            if (!kernels::isclassical(wfn_, get_qubit_position(q))
```

Comment:
Apologies for the naive question, but what does `kernels::isclassical` check? It looks from context like it's checking that the qubit is in a computational basis state?

Reply:
Yes, that's my understanding. I've added a new test to document that.